utils/fixed_decimal/src/decimal.rs

```diff
@@ -1015,17 +1066,96 @@ impl FixedDecimal {
             // 3b. Remove trailing zeros from self.digits to retain invariants
             // Note: this does not affect visible trailing zeros,
             // which is tracked by self.lower_magnitude
-            let position_past_last_nonzero_digit = self
+            let position_last_nonzero_digit = self
                 .digits
                 .iter()
                 .rposition(|x| *x != 0)
                 .map(|x| x + 1)
                 .unwrap_or(0);
-            self.digits.truncate(position_past_last_nonzero_digit);
+            self.digits.truncate(position_last_nonzero_digit);
+
             // 3c. By the invariant, there should still be at least 1 nonzero digit
             debug_assert!(!self.digits.is_empty());
+
+            let Some((last_digit, digits)) = self.digits.split_last_mut() else {
+                // Shouldn't be reachable anyways
+                #[cfg(debug_assertions)]
+                self.check_invariants();
+                return;
+            };
+
+            // 3d. Truncate to the next increment.
+            match increment {
+                RoundingIncrement::I1 => {
+                    // skip additional work if the increment is 1, since
+                    // it could be costly for the common case.
+                    #[cfg(debug_assertions)]
+                    self.check_invariants();
+                    return;
+                }
+                RoundingIncrement::I2 => {
+                    // Equivalent to (n / 2) * 2, which truncates to the next
+                    // multiple of two
+                    let num = *last_digit & 0xFE;
+
+                    if num == 0 {
+                        self.digits.pop();
+                    } else {
+                        *last_digit = num;
+                    }
+                }
+                RoundingIncrement::I5 => {
+                    if *last_digit < 5 {
+                        self.digits.pop();
+                    } else {
+                        *last_digit = 5;
+                    }
+                }
+                RoundingIncrement::I25 => {
+                    let Some((second_last_digit, _)) = digits.split_last_mut() else {
+                        // The number has no other digits aside from the last,
+                        // making it strictly less than 25.
```

Comment:
Wouldn't this happen for a number like `0.60`, which should not round to zero?

Reply:
Yep, you're right. I'll fix this bug after the exact API is decided, but in the meantime I'll add a test for it.

Reply:
Fixed.